app/src/main/java/com/github/wnder/GuessLocationActivity.java

```diff
@@ -351,7 +357,7 @@ private void updateCompassMode(){
     }
 
     private void compassModeSetup(){
-        TextView compassModeButtonView = (TextView) findViewById(R.id.compassMode);
+        View compassModeButtonView = findViewById(R.id.compassMode);
```

Comment:
There is a FloatingActionButton that you should probably use.

Reply:
done